src/main/java/Entities/UserDataClasses/HideableUserDataClasses/Courses.java

```diff
@@ -26,6 +27,16 @@ public Courses(Course course){
         super.data.add(course);
         this.data = super.data;
         this.hidden = true;
+    }
+    public Courses(String[] courseStrings){
+        super(new ArrayList<Course>());
+        for(String courseCodeData: courseStrings){
+            super.data.add(new Course(new CourseCode(courseCodeData)));
+        }
+        this.data = super.data;
+
```

Comment:
This constructor should probably include a precondition for "Each string in courseStrings is in the appropriate CourseCode format of abc123," because an error will occur if this condition is not met.